install/0000_90_cluster-version-operator_02_servicemonitor.yaml

```diff
@@ -57,6 +57,14 @@ spec:
         (time()-cluster_version_operator_update_retrieval_timestamp_seconds) >= 3600 and ignoring(condition, name, reason) cluster_operator_conditions{name="version", condition="RetrievedUpdates", endpoint="metrics", reason!="NoChannel"}
       labels:
         severity: warning
+    - alert: CurrentVersionNotVerified
+      annotations:
+        summary: The current version of the cluster is not verified.
+        description: The cluster was forcefully upgraded with a provided release image which verification failed, and therefore this cluster may not be trusted or may experience some issues. The verification process does not cover upgradeable checks that depend on the cluster state at the time when the update target was accepted.
+      expr: |
+        cluster_version{type="current", verified="false"}
+      labels:
+        severity: warning
```

Comment:
Wondering if it should be `info` but no strong feeling.

Reply:
Yes, I understand. There could be an argument for both of the severity labels.

In the said [Jira Card](https://issues.redhat.com/browse/OTA-187) it is suggested to have the `warning` label.

The [Openshift documentation](https://docs.openshift.com/container-platform/4.6/monitoring/managing-alerts.html) defines the labels as:

- **Warning**. The alert provides a warning notification about something that might require attention in order to prevent a problem from occurring. Warnings are typically routed to a ticketing system for non-immediate review.
- **Info**. The alert is provided for informational purposes only.

Resolving the issue of the cluster's version not being verified could prevent a problem from occurring, and it is rather important information.
However, the user can only apply such a version if an update is applied forcefully. And the information about the `--force` flag does state the potential risks. So the user that applied the upgrade knew about the risks, therefore a `warning` label maybe isn't necessary. 

I am leaning more toward the `warning` label, but I'll keep the discussion up and I will ask the team for opinions.

Reply:
@openshift/sre-alert-sme experts might have a view on this question :)

Reply:
tl;dr: this is too unactionable for a Warning, however there's definitive value for an Info alert providing additional context when investigating cluster issues.


From a fleet (managed openshift) point of view this alerts isn't really actionable for us SREs. The most we can do is send a servicelog message and silencing the alert on our end, but the msg would just be restating the contents of the alert ("please don't do that"). (For context: managed openshift SREs by default see Critical and Warning alerts from clusters except for the ones that we [explicitly chose not to route](https://github.com/openshift/configure-alertmanager-operator/blob/master/pkg/controller/secret/secret_controller.go#L160-L266).)

Using the [Alerting Consistency Enhancement Proposal](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/alerting-consistency.md) as a guide:
* **Warning** – The vast majority of alerts should use the severity. Issues at the warning level should be addressed in a timely manner, but don't pose an immediate threat to the operation of the cluster as a whole.
* **Info** – Info level alerts represent situations an administrator should be aware of, but that don't necessarily require any action. Use these sparingly, and consider instead reporting this information via Kubernetes events.

I'd say in this context this falls into the category of "Info". This alert on its own doesn't really require any action, as the cluster admin explicitly chose to do an upgrade like this and if there are no issues with a cluster running like this, there's also no need for action to remedy this situation.

This alert's usefulness is in the additional context (information) that it provides – if there __is__ something wrong with the cluster and Critical/Warning alerts are firing, then you as a cluster admin/investigating SRE will be glad that this alert's there to point at the unverified upgrade being a potential culprit.

Reply:
+1 to @mmazur ! execellent dive into this. We should also look to automate messages to customer on this as well :)

Reply:
Thank you for the great insights! I'll change the severity to `info`.